### PR TITLE
Feature/automatic testing

### PR DESCRIPTION
**Pull request by saniemi**
A new wrapper code with some helper functions to facilitate automated testing.

### CODE REVIEW

File: DataScience/Analytics/prototype/automatic_testing.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    :type df: pandas.DataFrame
+    :param methods: a tuple listing methods to analyse
+    :type methods: tuple
+
+    :return: results of the performance computations
+    :rtype: dict
+    """
+    # simple performance metrics that can be computed directly from the data frame and dummies
+    addresses = len(df.index)
+    linked = len(df.loc[~df['UPRN_PROTO'].isnull()].index)
```
Comment (Mironor) [inline DataScience/Analytics/prototype/automatic_testing.py]: `len(df.loc[~df['UPRN_PROTO'].isnull()].index)` this is far from readable

File: DataScience/Analytics/prototype/automatic_testing.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    """
+    # query data and place it to a Pandas DataFrame
+    data = _get_data_from_db('select * from results;')
+
+    # convert date to datetime
+    data['date'] = pd.to_datetime(data['date'])
+
+    columns_to_plot = ['addresses', 'correct', 'false_positive', 'linked', 'new_UPRNs', 'not_linked']
+
+    # create figures
```
Comment (Mironor) [inline DataScience/Analytics/prototype/automatic_testing.py]: This is a kind of comment that would better be a private function named `_createFigures`

File: DataScience/Analytics/prototype/automatic_testing.py (hunk 1 above)
Comment (saniemi) [inline DataScience/Analytics/prototype/automatic_testing.py]: Simplified the logic.

File: DataScience/Analytics/prototype/automatic_testing.py (hunk 2 above)
Comment (saniemi) [inline DataScience/Analytics/prototype/automatic_testing.py]: Good idea, moved the figure creation to a private function.

**Pull request merged**